server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -1465,6 +1503,59 @@ def _add_feature_stats(
         model_endpoint_object.spec.feature_stats = feature_stats
         return model_endpoint_object, model_obj
 
+    @staticmethod
+    def _get_mep_stats_dict_from_parquet(
+        db_session,
+        project,
+        uid,
+        kind,
+    ) -> tuple[dict, typing.Optional[datetime]]:
+        parquet_target = (
+            services.api.crud.model_monitoring.helpers.get_monitoring_parquet_path(
+                db_session=db_session,
+                project=project,
+                kind="parquet_stats",
+            )
+        )
+
+        target = ParquetTarget(
+            path=parquet_target,
+            partition_cols=[WriterEvent.ENDPOINT_ID, mm_constants.StatsData.STATS_NAME],
+        )
+        try:
+            df = target.as_df(
+                additional_filters=[
+                    (WriterEvent.ENDPOINT_ID, "=", uid),
+                    (mm_constants.StatsData.STATS_NAME, "=", kind),
+                ]
+            )
+        except Exception as exc:
+            logger.warning(
+                "Failed to read stats from parquet, you may need to check the parquet file",
+                project=project,
+                endpoint_id=uid,
+                kind=kind,
+                error=mlrun.errors.err_to_str(exc),
+            )
+            return {}, None
```

Comment:
Needs a bit vertical spacing for readability.
```suggestion
            return {}, None
            
```